### PR TITLE
feat!: more methods of `struct` namespace

### PR DESCRIPTION
**Pull request by etiennebacher**

### CODE REVIEW

File: R/expr-struct.R [hunk 1]
```diff
@@ -34,25 +65,103 @@ expr_struct_field <- function(...) {
...
+#' )
+expr_struct_rename_fields <- function(names) {
   wrap({
-    check_dots_unnamed()
-
-    dots <- list2(...) |>
-      unlist(recursive = FALSE)
-    check_character(dots, arg = "...", allow_na = FALSE)
-
-    self$`_rexpr`$struct_rename_fields(dots)
```
Comment (eitsupi) [inline R/expr-struct.R:45]: Why?

**Review by eitsupi — CHANGES_REQUESTED** [review]

File: R/expr-struct.R (hunk 1 above)
Comment (etiennebacher) [reply R/expr-struct.R:45]: It's not `*` in py-polars, it's a vector: https://docs.pola.rs/api/python/dev/reference/expressions/api/polars.Expr.struct.rename_fields.html

And the input in Rust is `StringSexp` so I don't see why we'd use `...` here

Comment (eitsupi) [reply R/expr-struct.R:45]: I see. In that case, please use the check function to make sure there is no NA.

Comment (etiennebacher) [reply R/expr-struct.R:45]: Done

**Pull request merged**